### PR TITLE
DAOS-7270 dfuse: Add notification that IL is working.

### PR DESCRIPTION
**Pull request by ashleypittman**
Have the interception library print to stderr when it intercepts and on teardown.
This is controlled via a D_IL_REPORT environment variable, set it to 0 for just
the teardown report, or a positive number to have it log on the first intercepted
read and write calls for confirmation that it's working correctly.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
You can see this working in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6038/12/ but I've pushed a new version to disable the intercept test again as it fails for other reasons.  This PR is ready for review however.

**Review by omar-ocampo — APPROVED** [review]

File: src/tests/ftest/util/ior_test_base.py [hunk 1]
```diff
@@ -238,7 +238,12 @@ def run_ior(self, manager, processes, intercept=None, display_space=True,
         """
         env = self.ior_cmd.get_default_env(str(manager), self.client_log)
         if intercept:
-            env["LD_PRELOAD"] = intercept
+            env['LD_PRELOAD'] = intercept
+            env['D_LOG_MASK'] = 'INFO'
+            env['D_IL_REPORT'] = '1'
+            #env['D_LOG_MASK'] = 'INFO,IL=DEBUG'
+            #env['DD_MASK'] = 'all'
+            #env['DD_SUBSYS'] = 'all'
```
Comment (daltonbohning) [inline src/tests/ftest/util/ior_test_base.py:246]: Are these lines (DD_MASK and DD_SUBSYS) meant to be left here?

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/util/ior_test_base.py (hunk 1 above)
Comment (ashleypittman) [reply src/tests/ftest/util/ior_test_base.py:246]: Yes, it took me ages to find out how to enable debug in this test.

**Review by jolivier23 — APPROVED** [review]

**Review by daltonbohning — DISMISSED** [review]

File: src/client/dfuse/il/int_posix.c [hunk 2]
```diff
@@ -280,6 +291,26 @@ ioil_init(void)
...
+
+	if (ioil_iog.iog_file_count == 0)
+		return;
+
+	summary = getenv("D_IL_REPORT");
+
+	if (summary == NULL)
+		return;
+
+	fprintf(stderr, "Performed %"PRIu64" reads and %"PRIu64" writes from %"PRIu64" files\n",
```
Comment (johannlombardi) [inline src/client/dfuse/il/int_posix.c:310]: I think that we need a sort of header here to show that this comes from the interception library. Maybe something like [libioil] ...

File: src/client/dfuse/il/int_posix.c [hunk 3]
```diff
@@ -291,6 +322,8 @@ ioil_fini(void)
 	DFUSE_TRA_DOWN(&ioil_iog);
 	vector_destroy(&fd_table);
 
+	ioil_show_summary();
```
Comment (johannlombardi) [inline src/client/dfuse/il/int_posix.c:325]: If we have a very long running job, we are going to have to wait until the end to know whether the IL is really working. My original idea was to have a number in this env variable and print to stderr (with a header to differentiate from regular APP output) the first N successful interception and then a summary at the end.

File: src/client/dfuse/il/int_posix.c [hunk 4]
```diff
@@ -280,6 +291,26 @@ ioil_init(void)
...
+{
+	char *summary;
+
+	D_INFO("Performed %"PRIu64" reads and %"PRIu64" writes from %"PRIu64" files\n",
+	       ioil_iog.iog_read_count, ioil_iog.iog_write_count, ioil_iog.iog_file_count);
+
+	if (ioil_iog.iog_file_count == 0)
+		return;
+
+	summary = getenv("D_IL_REPORT");
```
Comment (johannlombardi) [inline src/client/dfuse/il/int_posix.c:305]: This can be done in a follow-up doc-only patch, but we *really* need to document that in the manual.

**Review by johannlombardi — CHANGES_REQUESTED** [review]
Please see comments in the code.

**Review by johannlombardi — DISMISSED** [review]

**Review by daltonbohning — DISMISSED** [review]

File: src/client/dfuse/il/int_posix.c [hunk 5]
```diff
@@ -265,6 +298,13 @@ ioil_init(void)
 		return;
 	}
 
+	d_getenv_int("D_IL_REPORT", &report_count);
+	if (report_count != -1) {
+		printf("Will report %d\n", report_count);
```
Comment (ashleypittman) [inline src/client/dfuse/il/int_posix.c:303]: This is a debug line I left in.

**Pull request merged**